### ISSUE TITLE
Update to version 4.0.0 and the method of connect() will not return value until the await interrupt terminates

### ISSUE THREAD

**Issue by natezhengbne** [shamblett/mqtt_client#49]
Version: 4.0.0 (MqttClient _client) -> It could not await the MqttClient.connect return
```
MqttClientConnectionStatus cs = await _client.connect();
if(cs != null && cs.state == ConnectionState.connected){
```
Version: 3.3.3 -> It's working pretty well
```
ConnectionState cs = await _client.connect();
 if(cs != null && cs == ConnectionState.connected){
```

**Comment by shamblett** [comment]
OK, this may be connected with[ issue 50](https://github.com/shamblett/mqtt_client/issues/50), is it possible for you to import the client with a prefix, e.g. 'as mqtt' then change the code to use the prefix, i.e.
mqtt.MqttClientConnectionStatus etc and see if the fault persists.

**Comment by shamblett** [comment]
Client re-published at 5.0.0, please retest
